### PR TITLE
Add x25519 test cases

### PR DESCRIPTION
**Pull request by sohhass**
Hey folks!

This adds X25519 test cases for key generation and shared secret computation. The shared secret computation test vectors are taken from RFC 7748 section 6.1.

Corresponding optee_os PR #5297

### CODE REVIEW

File: host/xtest/regression_4000.c [hunk 1]
```diff
@@ -5519,4 +5565,193 @@ static void xtest_tee_test_4014(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 4014, xtest_tee_test_4014,
 		"Test SM2 KEP (key derivation)");
+
+static void xtest_tee_test_4015(ADBG_Case_t *c)
+{
+	TEEC_Session session = { };
+	uint32_t ret_orig = 0;
+	TEE_OperationHandle op = TEE_HANDLE_NULL;
+	TEE_ObjectHandle keyAllice = TEE_HANDLE_NULL;
```
Comment (jforissier) [inline host/xtest/regression_4000.c:5574]: Miss-spelling: `Alice`. Prefer lowercase names for variables: `key_alice`, `key_bob`.

**Review by jforissier — APPROVED** [review]
LGTM with one minor comment below.

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: host/xtest/regression_4000.c (hunk 1 above)
Comment (sohhass) [reply host/xtest/regression_4000.c]: > Miss-spelling: `Alice`. Prefer lowercase names for variables: `key_alice`, `key_bob`.

Done!

File: host/xtest/regression_4000.c [hunk 2]
```diff
@@ -4698,6 +4716,34 @@ static void xtest_tee_test_4007_ecc(ADBG_Case_t *c)
...
+	TEE_Attribute params[4] = { };
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+	xtest_teec_open_session(&session, &crypt_user_ta_uuid, NULL,
+				&ret_orig)))
+		return;
+
+	Do_ADBG_BeginSubCase(c, "Generate X25519 key");
+
+	param_count = 0;
```
Comment (etienne-lms) [inline host/xtest/regression_4000.c:4733]: not needed

File: host/xtest/regression_4000.c [hunk 3]
```diff
@@ -5519,4 +5565,193 @@ static void xtest_tee_test_4014(ADBG_Case_t *c)
...
+
+static void xtest_tee_test_4015(ADBG_Case_t *c)
+{
+	TEEC_Session session = { };
+	uint32_t ret_orig = 0;
+	TEE_OperationHandle op = TEE_HANDLE_NULL;
+	TEE_ObjectHandle key_alice = TEE_HANDLE_NULL;
+	TEE_ObjectHandle key_bob = TEE_HANDLE_NULL;
+	TEE_ObjectHandle sv_handle = TEE_HANDLE_NULL;
+	TEE_Attribute params[9] = { };
```
Comment (etienne-lms) [inline host/xtest/regression_4000.c:5577]: Only 2 attributes are loaded at most. Use 2 instead of 9 here?

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>` with minor indentation comments preferred to be addressed.

File: host/xtest/regression_4000.c [hunk 4]
```diff
@@ -4698,6 +4716,31 @@ static void xtest_tee_test_4007_ecc(ADBG_Case_t *c)
 ADBG_CASE_DEFINE(regression, 4007_ecc, xtest_tee_test_4007_ecc,
 		"Test TEE Internal API Generate ECC key");
 
+static void xtest_tee_test_4007_x25519(ADBG_Case_t *c)
+{
+	TEEC_Session session = { };
+	uint32_t ret_orig = 0;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c,
+	xtest_teec_open_session(&session, &crypt_user_ta_uuid, NULL,
```
[the remark this answers is not in the document]
Comment (sohhass) [reply host/xtest/regression_4000.c]: Thanks for the review, comments addressed!

**Pull request merged**